default/scripting/buildings/GAS_GIANT_GEN.focs.txt

```diff
@@ -16,29 +16,36 @@ BuildingType
             scope = And [
                 Planet
                 InSystem id = Source.SystemID
-                Planet type = GasGiant
```

Comment:
What's the point of these changes? You're removing the scope conditions that limit the effects to gas giants or non-gas giants, and adding conditions that the building only functions when it is contained by a planet with no population or with population. The building should always be on a planet, so that part of the activation is redundant... But I'm confused by changing the scope discrimination between gas giant and other, to an activation condition based on population or not.

Reply:
I reversed the application:
Before the scope for the +10/+5 effect was differentiating between gas giants and other planets. 
Now the source makes the difference for all planets.

So in effect, if there is no unpopulated gas giant generator (+10 bonus), a populated gas giant generator adds +5 
Before only the populated gas giant had only +5, the other planets had the +10 bonus,.

About Planet condition in the containedBy in the activation: the building is also contained by a system, isnt it?

Reply:
Yes; systems contain fleets, ships, planets, buildings, and possibly fields. Fleets contain ships, and planets contain buildings.

Reply:
I was afraid that without the Planet condition, the ContainedBy would put the system and the planet in the scope. Not sure what the Population condition does with Systems, but if it isnt ignored it would do the wrong thing in one of the EffectGroups.

Reply:
`ContainedBy Planet` will match things contained by planets, not things contained by some system... `Contains Planet` would match things that contain a planet, such as a system.

Reply:
exactly yes, thats why i added the Planet condition to the ContainedBy, because you asked

quoting you: "adding conditions that the building only functions when it is contained by a planet with no population or with population. The building should always be on a planet, so that part of the activation is redundant..."

Reply:
Buildings are always on a planet. Therefore, checking in the activation condition if the building (which is what is being tested by an activation condition) is on a planet is pointless.

This has nothing to do with the scope.

Reply:
Oh right

Reply:
Got confused ... activation not scope..
..but...
the activation checks not if the building is contained by a planet

the activation checks if the building containing the a planet is populated or not.

i would think that one the following would evaluate to true for the containing system regardless of the containing planet

> activation = ContainedBy Not Population high = 0

> activation = ContainedBy Population high = 0

Reply:
Systems don't have population. But OK, if you want to filter out the system before testing the population, fine.

Reply:
Thanks for you patience geoff, I figured out why the problem i was referring to does not exist in this case (The +10 bonus has priority, so I dont have to constrain the activation condition for the +5), so I could simplify the activation condition even more (Source instead of ContainedBy Not Population high = 0).

Reply:
I'd actually keep the activation conditions to clarify the intent better than just having the stacking groups and relying on the ordering of effects in the script.